lib/puppet/type/ini_setting.rb

```diff
@@ -39,6 +39,11 @@ def munge_boolean_md5(value)
     end
   end
 
+  newparam(:force_new_section_creation, :boolean => true, :parent => Puppet::Parameter::Boolean) do
+    desc 'Create setting only if the section exists'
+    defaultto(true)
```

Comment:
Defaulting to true seems like the opposite behavior of what currently exists?

Reply:
The current behavior is that it creates the section and the setting if it does not exist in the ini file.  So defaulting force_new_section_creation to True will also create the new section.